src/Service.Tests/Authorization/AuthorizationResolverUnitTests.cs

```diff
@@ -1204,12 +1190,167 @@ public void ValidateClientRoleHeaderClaimIsAddedToClaimsInRequestContext()
+        }
+
+        /// <summary>
+        /// Validates that AuthorizationResolver.GetProcessedUserClaims(httpContext) does not resolve claims sourced from
+        /// an unauthenticated ClaimsIdentity object within a ClaimsPrincipal.
+        /// While no scenarios currently inject an unauthenticated ClaimsIdentity object into a ClaimsPrincipal,
+        /// any scenarios that do so will have their unauthenticated ClaimsIdentity claims ignored.
+        /// </summary>
+        [TestMethod]
+        public void ValidateUnauthenticatedUserClaimsAreNotResolvedWhenProcessingUserClaims()
+        {
+            // Arrange
+            Mock<HttpContext> context = new();
+
+            // Create authenticated ClaimsIdentity object with list of claims.
+            List<Claim> authenticatedUserclaims = new()
+            {
+                new("scp", "openid", ClaimValueTypes.String),
+                new(AuthenticationOptions.ROLE_CLAIM_TYPE, TEST_ROLE, ClaimValueTypes.String),
+            };
+
+            //Add identity object to the Mock context object.
+            ClaimsIdentity authenticatedIdentity = new(TEST_AUTHENTICATION_TYPE, TEST_CLAIMTYPE_NAME, AuthenticationOptions.ROLE_CLAIM_TYPE);
+            authenticatedIdentity.AddClaims(authenticatedUserclaims);
+
+            // Create unauthenticated ClaimsIdentity object with list of claims.
+            List<Claim> unauthenticatedClaims = new()
+            {
+                new("scp", "invalidScope", ClaimValueTypes.String),
+                new(AuthenticationOptions.ROLE_CLAIM_TYPE, "Don't_Parse_This_Role",ClaimValueTypes.String)
+            };
+
+            // ClaimsIdentity is unauthenticated because the authenticationType is null.
+            ClaimsIdentity unauthenticatedIdentity = new(authenticationType: null, TEST_CLAIMTYPE_NAME, AuthenticationOptions.ROLE_CLAIM_TYPE);
+            unauthenticatedIdentity.AddClaims(unauthenticatedClaims);
+
+            //Add identities object to the Mock context object.
+            ClaimsPrincipal principal = new();
+            principal.AddIdentity(authenticatedIdentity);
+            principal.AddIdentity(unauthenticatedIdentity);
+
+            context.Setup(x => x.User).Returns(principal);
+            context.Setup(x => x.Request.Headers[AuthorizationResolver.CLIENT_ROLE_HEADER]).Returns(TEST_ROLE);
+
+            // Act
+            Dictionary<string, string> claimsInRequestContext = AuthorizationResolver.GetProcessedUserClaims(context.Object);
+
+            // Assert
+            Assert.AreEqual(expected: authenticatedUserclaims.Count, actual: claimsInRequestContext.Count, message: "Only two claims should be present.");
+            Assert.AreEqual(expected: "openid", actual: claimsInRequestContext["scp"], message: "Unexpected scp claim returned.");
+            Assert.AreEqual(expected: TEST_ROLE, actual: claimsInRequestContext[AuthenticationOptions.ROLE_CLAIM_TYPE], message: "Unexpected roles claim returned.");
```

Comment:
`roles` claim seems to be very special -> because it will always have a single role even if the there were 2 authenticated identities. So, this test doesnt seem to point out the distinction between authenticated v/s unauthenticated. Can we check a regular claim `foo`, if it gets added twice like a JSON array?

Reply:
Updated. added an `oid` claim to both identities and asserted that the oid claim value used for the unauthenticated identity was NOT resolved. Also updated the test for 'roles' claim (even if special) to validate that the unauthenticated roles claim was not used instead of checking that the authenticated role claim was used. While essentially the same test, the updated version aligns with the test name explicitly.